bigtable-dataflow-import/src/main/java/com/google/cloud/bigtable/sequencefile/HBaseImportIO.java

```diff
@@ -0,0 +1,138 @@
+import com.google.cloud.dataflow.sdk.runners.DirectPipelineRunner;
+import com.google.cloud.dataflow.sdk.transforms.PTransform;
+import com.google.cloud.dataflow.sdk.transforms.ParDo;
+import com.google.cloud.dataflow.sdk.values.KV;
+import com.google.cloud.dataflow.sdk.values.PCollection;
+import com.google.cloud.hadoop.fs.gcs.GoogleHadoopFS;
+import com.google.cloud.hadoop.fs.gcs.GoogleHadoopFileSystem;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableMap;
+
+import org.apache.hadoop.hbase.client.Mutation;
+import org.apache.hadoop.hbase.client.Result;
+import org.apache.hadoop.hbase.io.ImmutableBytesWritable;
+import org.apache.hadoop.hbase.mapreduce.ResultSerialization;
+import org.apache.hadoop.io.serializer.WritableSerialization;
+import org.apache.hadoop.mapreduce.lib.input.SequenceFileInputFormat;
+
+import java.util.Map;
+
+/**
+ * <p>
+ * {@link com.google.cloud.dataflow.sdk.io.Source} and
+ * {@link com.google.cloud.dataflow.sdk.transforms.PTransform}s for importing HBase Sequence Files
+ * into <a href="https://cloud.google.com/bigtable/">Google Cloud Bigtable</a>.
+ * <p>
+ * User of this class should provide dependency for
+ * {@code org.apache.hadoop.hbase.mapreduce.ResultSerialization}, which may be found in
+ * hbase-server jars.
+ *
+ * The following code snippets sets up and runs an import pipeline:
+ *
+ * <pre>
+ * {@code
+ *   HBaseImportOptions options =
+ *       PipelineOptionsFactory.fromArgs(args).withValidation().as(HBaseImportOptions.class);
+ *   Pipeline p = CloudBigtableIO.initializeForWrite(Pipeline.create(options));
+ *   p
+ *       .apply("ReadSequenceFile", Read.from(HBaseImportIO.createSource(options)))
+ *       .apply("ConvertResultToMutations", HBaseImportIO.transformToMutations())
+ *       .apply("WriteToTable", CloudBigtableIO.writeToTable(
+ *           CloudBigtableTableConfiguration.fromCBTOptions(options)));
+ *   p.run();
+ * }
+ * </pre>
+ */
+public class HBaseImportIO {
+  // Needed for HBase 0.94 format. Copied from ResultSerialization.IMPORT_FORMAT_VER.
+  @VisibleForTesting
+  static final String IMPORT_FORMAT_VER = "hbase.import.version";
+  @VisibleForTesting
+  static final String VERSION_094_STRING = "0.94";
+  @VisibleForTesting
+  static final String IO_SERIALIZATIONS = "io.serializations";
+  @VisibleForTesting
+  static final String FS_GS_IMPL = "fs.gs.impl";
+  @VisibleForTesting
+  static final String FS_ABSTRACT_FILESYSTEM_GS_IMPL = "fs.AbstractFileSystem.gs.impl";
+  @VisibleForTesting
+  static final String FS_GS_PROJECT_ID = "fs.gs.project.id";
+  @VisibleForTesting
+  static final String CLOUD_SVC_ACCOUNT_AUTH_ENABLED = "google.cloud.auth.service.account.enable";
+  @VisibleForTesting
+  static final String CLOUD_SVC_ACCOUNT_EMAIL = "google.cloud.auth.service.account.email";
+  @VisibleForTesting
+  static final Map<String, String> COMMON_SF_SERIALIZER_PROPERTIES = ImmutableMap.of(
```

Comment:
DEAFULT_... ? This looks like a set of default values.
What is "SF", I see "FS" in some of the field names. I know these are internal field names, still want to them to be more readable. If you like to use the abbreviation, maybe add some comments? Thanks!